src/main/java/com/arpnetworking/metrics/mad/parsers/CollectdJsonToRecordParser.java

```diff
@@ -0,0 +1,418 @@
+/**
+ * Copyright 2016 Smartsheet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.mad.parsers;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.commons.jackson.databind.ObjectMapperFactory;
+import com.arpnetworking.logback.annotations.Loggable;
+import com.arpnetworking.metrics.common.parsers.Parser;
+import com.arpnetworking.metrics.common.parsers.exceptions.ParsingException;
+import com.arpnetworking.metrics.mad.model.DefaultMetric;
+import com.arpnetworking.metrics.mad.model.DefaultRecord;
+import com.arpnetworking.metrics.mad.model.Metric;
+import com.arpnetworking.metrics.mad.model.Record;
+import com.arpnetworking.tsdcore.model.MetricType;
+import com.arpnetworking.tsdcore.model.Quantity;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.module.afterburner.AfterburnerModule;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import net.sf.oval.constraint.CheckWith;
+import net.sf.oval.constraint.CheckWithCheck;
+import net.sf.oval.constraint.NotNull;
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Parses Collectd JSON data as a {@link Record}.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public final class CollectdJsonToRecordParser implements Parser<List<DefaultRecord.Builder>> {
+    /**
+     * Parses a collectd POST body.
+     *
+     * @param data byte array of json encoded data
+     * @return A list of {@link DefaultRecord.Builder}
+     * @throws ParsingException if the body is not parsable as collectd formatted json data
+     */
+    @SuppressFBWarnings("SIC_INNER_SHOULD_BE_STATIC_ANON")
+    public List<DefaultRecord.Builder> parse(final byte[] data) throws ParsingException {
+        try {
+            final List<CollectdRecord> records = OBJECT_MAPPER.readValue(data, new TypeReference<List<CollectdRecord>>() { });
```

Comment:
This should be a private static final instance; this should address the FB warning above.